highfidelity/hifi
# Case 17368: Fix wearables not disappearing with avatar

https://highfidelity.manuscript.com/f/cases/17368/When-going-away-pressing-escape-key-wearables-do-not-disappear

#9462

labels: open source, engine, pre-merge testing required

Test plan:
- Run [this script](https://gist.githubusercontent.com/SamGondelman/f18015d6dcf95c3cdf4eb02f5bc515fa/raw/80437e30f2e5848b86c69543a609883f676950cf/AwayToggleWearablesTest.js) in a zone with shadows enabled, you'll see a bunch of entities (1 box, two models, one text, one image, one web, one particle, one polyline, one polyvox, one grid, one ring gizmo, one material).  Some will have shadows, some won't, this is expected, as it is in master.  Just make sure that when objects disappear during this test, their shadows do too.
- Toggle your avatar's visibility by pressing Escape. Your avatar will disappear but the entities will remain the same.  Move around to reappear.
- Press "h". The entities are now attached to your avatar. They should appear the same, but they'll follow you when you move. (Note: the material entity will vanish because its parent has been set, this is fine, ignore it for the rest of the test)
- Toggle your avatar's visibility by pressing Escape. Your avatar AND all the entities will appear/disappear at the same time (as will their shadows).
- While visible, press "h" again. The entities won't follow you anymore and won't disappear when you press escape.
